### DISCUSSION TITLE
Why keep two decimal places when exporting?

### DISCUSSION THREAD

**Discussion by boomsi** [fabricjs/fabric.js#9438] [Q&A] [answered]
The toFixed operation is causing changes in the position and size of visual elements during export and import.

So why?

<img width="1097" alt="image" src="https://github.com/fabricjs/fabric.js/assets/39713960/8e7a3121-9434-41ab-b73a-759c0a682a8a">

**Comment by ShaMan123** [comment, marked as answer]
This is for reducing storing space
I agree it should not be on by default

**Reply by boomsi** [reply]
Oh,thinks

**Comment by asturur** [comment]
is a matter of compativility with old tests too. The default should be like 8 not 2, but not unlimited because then you get really large jsons

**Reply by ShaMan123** [reply]
> but not unlimited 

I disagree
An optimization like this is an opt in used only for storing, the first expectation is for rendering to be intact

**Reply by gloriousjob** [reply]
I think 8 would be a good balance for a default for the NUM_FRACTION_DIGITS. It sounds like v6 has it at 4.
It looks like this is targeting exports only. So @ShaMan123, are you suggesting these should be changed to where NUM_FRACTION_DIGITS is undefined/-1, it would ignore using toFixed but if defined, then limit? My vote would be default to 6-8 but I could see allowing the dev to have the option to not use toFixed (really, not toFixed but something that doesn't add extra 0's if they aren't needed. I'd rather have 8 than 8.00000000).

**Reply by ShaMan123** [reply]
exactly
